core/utils.ts

```diff
@@ -116,20 +117,34 @@ export function validate(compiledGraph: dataform.ICompiledGraph): dataform.IGrap
   allActions.forEach(action => {
     if (allActions.filter(subAction => subAction.name == action.name).length > 1) {
       const actionName = action.name;
-      const message = `Duplicate action name detected, names must be unique across tables, assertions, and operations: "${action.name}"`;
+      const message = `Duplicate action name detected. Names within a schema must be unique across tables, assertions, and operations: "${
+        action.name
+      }"`;
       validationErrors.push(dataform.ValidationError.create({ message, actionName }));
     }
   });
 
   const actionsByName: { [name: string]: dataform.IExecutionAction } = {};
   allActions.forEach(action => (actionsByName[action.name] = action));
 
-  // Check all dependencies actually exist.
+  // Check all dependencies actually exist and are not ambiguous.
   allActions.forEach(action => {
     const actionName = action.name;
     (action.dependencies || []).forEach((dependency: string) => {
-      if (allActionNames.indexOf(dependency) < 0) {
-        const message = `Missing dependency detected: Node "${action.name}" depends on "${dependency}" which does not exist.`;
+      const [matchedDep, err] = matchFQName(dependency, allActionNames);
```

Comment:
i don't think we need to do any matching here, right? my understanding of the code in `session.ts` is that it now fully-resolves dependencies. all we should need to do is something like:

```
if (!allActionNames.includes(dependency)) {
  // add a validation error
}
```

Reply:
That's correct. Good catch!